akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala

```diff
@@ -220,4 +220,11 @@ object Sink extends SinkApply {
     new Sink(new AcknowledgeSink(bufferSize, DefaultAttributes.acknowledgeSink, shape("AcknowledgeSink"), timeout))
   }
 
+  /**
+   * Crates a `Sink` that is materialized into `Source` to connect with another flow.
+   * This sink gets backpressure from second flow and does not have internal buffer.
+   */
+  def pipedSink[T](): Sink[T, Source[T, Unit]] =
+    Sink.publisher[T].mapMaterializedValue(Source(_)).named("PipedSink")
```

Comment:
> I.e. if your code needs a Source, then your code should not accept a Source as a parameter to a function, but return a Sink instead.

Either that is wrong, or Akka HTTP is going against the flow of how Akka streams is meant to be used, since Akka Streams accepts Sources as parameters all over the place, for client request body streaming, for server response body streaming, in WebSockets, everywhere.  I just don't think the ideals being expressed here are at all consistent with the reality of providing a usable API.